device/core/src/client.ts

```diff
@@ -674,7 +441,7 @@ export class Client extends EventEmitter {
 
   private _addMethodCallback(methodName: string, callback: (request: DeviceMethodRequest, response: DeviceMethodResponse) => void): void {
     const self = this;
-    (this._receiver as DeviceMethodReceiver).onDeviceMethod(methodName, (message) => {
+    (this._transport as DeviceMethodTransport).onDeviceMethod(methodName, (message) => {
```

Comment:
>(this._transport as DeviceMethodTransport). [](start = 4, length = 43)

What will this do if the transport doesn't implement the DeviceMethodTransport interface?  Is that even possible?  Is the correct exception thrown?

Reply:
it's a little convoluted but the test for device methods support is in _validateDeviceMethodInputs. so an exception will be thrown alright, just from a weird place I guess.

---
In reply to: [145240275](https://github.com/Azure/azure-iot-sdk-node/pull/109#discussion_r145240275) [](ancestors = 145240275)

Reply:
I've modified the code to actually throw a NotImplementedError instead of a generic error though. thanks for pointing this out.

---
In reply to: [145266360](https://github.com/Azure/azure-iot-sdk-node/pull/109#discussion_r145266360) [](ancestors = 145266360,145240275)